server/sonar-server/src/test/java/org/sonar/server/plugins/ws/PendingPluginsWsActionTest.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.plugins.ws;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.server.plugins.PluginDownloader;
+import org.sonar.server.plugins.ServerPluginJarsInstaller;
+import org.sonar.server.ws.WsTester;
+
+import static com.google.common.collect.ImmutableList.of;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.sonar.test.JsonAssert.assertJson;
+
+public class PendingPluginsWsActionTest {
+
+  private static final String DUMMY_CONTROLLER_KEY = "dummy";
+
+  @Mock
+  private PluginDownloader pluginDownloader;
+  @Mock
+  private ServerPluginJarsInstaller serverPluginJarsInstaller;
+  @InjectMocks
+  private PendingPluginsWsAction underTest;
+
+  @Mock
+  private Request request;
+  private WsTester.TestResponse response = new WsTester.TestResponse();
+
+  @Before
+  public void createAndWireMocks() throws Exception {
+    initMocks(this);
```

Comment:
I agree it is more concise.

I wrote it that way initially but then I changed my mind.

When I have a bunch of properties at the beginning of the test class, I find annotations much more readable that pure Java code, and yes, at the cost of the a little reflection.

I couldn't read my own code, so, out of pure kindness to my fellow developers, I went that in that direction instead of adding useless empty lines of moving constants to the end of the file :)